vault/resource_kubernetes_secret_backend_role.go

```diff
@@ -200,6 +218,7 @@ func kubernetesSecretBackendRoleRead(_ context.Context, d *schema.ResourceData,
 	fields := []string{
 		consts.FieldName,
 		fieldAllowedKubernetesNamespaces,
+		fieldAllowedKubernetesNamespaceSelector,
```

Comment:
If ran against a Vault server version less than 1.12 `resp.Data["allowed_kubernetes_namespace_selector"]` will be nil and we will set that to the TF state file. I think we should skip setting this field. 

In the for loop below you could do something like

```go
for _, k := range fields {
	if field == fieldAllowedKubernetesNamespaceSelector && !provider.IsAPISupported(meta, provider.VaultVersion112) {
		continue
	}
// d.Set ...
```

Reply:
Thanks, this also looks better than handling this outside of the loop, I applied this also in the `CreateUpdate` function.